### PR TITLE
Removed MonadError[Eval, Throwable]

### PR DESCRIPTION
**Pull request by djspiewak**
Fixes a couple outstanding issues.  Also obsoletes #1634.  See #1661 for an in-depth discussion of why this is necessary.

### CODE REVIEW

**Comment by adelbertc** [comment]
👍

**Comment by kailuowang** [comment]
Shall we address @alexandru's [objection here](https://github.com/typelevel/cats/issues/1661#issuecomment-301039355) first?

**Comment by ceedubs** [comment]
I'm in favor of this change. However, I think that we should give a little more time for discussion on #1661. I propose that we allow a week for discussion and go with what the general consensus seems to be after that amount of time.

I also think it would be great if @non could weigh in, since I think that he originally added this code.

File: tests/src/test/scala/cats/tests/EvalTests.scala [hunk 1]
```diff
@@ -83,60 +82,13 @@ class EvalTests extends CatsSuite {
...
 
-    {
-      // we need exceptions which occur during .value calls to be
-      // equal to each other (equivalent behavior).
-      implicit def eqWithTry[A: Eq]: Eq[Eval[A]] =
-        Eq[Try[A]].on((e: Eval[A]) => Try(e.value))
-
-      checkAll("Eval[Int]", MonadErrorTests[Eval, Throwable].monadError[Int, Int, Int])
-    }
+    checkAll("Eval[Int]", MonadTests[Eval].monad[Int, Int, Int])
```
Comment (ceedubs) [inline tests/src/test/scala/cats/tests/EvalTests.scala]: Aren't these `MonadTests` just a subset of the `BimonadTests` that are happening above?

Also it seems a little off to me that the `SerializableTests` for the `Bimonad` instance are below in a separate scope. This was the case before your PR I think, but I don't know if there's a good reason for it.

**Comment by kailuowang** [comment]
quick note, the conflict is due to the merge of #1634.

**Comment by djspiewak** [comment]
@kailuowang Apparently I had already fixed the conflict, but forgot to push.  Pushed now!

**Comment by kailuowang** [comment]
Thanks @djspiewak I think we've allowed 10 days of time for people to weigh in. Looks like that the only thing holding is this [comment](https://github.com/typelevel/cats/pull/1663#discussion_r117142358) by @ceedubs ?

File: tests/src/test/scala/cats/tests/EvalTests.scala (hunk 1 above)
Comment (djspiewak) [inline tests/src/test/scala/cats/tests/EvalTests.scala]: You're correct that `MonadTests` should just be a subset of `BimonadTests`.  Similar to the `SerializableTests`, I'm sort of puzzled as to why this was the case before.  Mostly I was just carrying it along.  If we're all in agreement that it looks fishy though, I'll just zap this line.

Comment (kailuowang) [inline tests/src/test/scala/cats/tests/EvalTests.scala]: I 👍  especially if it's confirmed that we don't have a coverage drop after.

**Comment by ceedubs** [comment]
There seems to be general (though unfortunately not quite unanimous) consensus on this in #1661, after allowing 10 days or so for discussion. Merging with 4 approvals.

**Pull request merged**